torch/cuda/nccl.py

```diff
@@ -42,15 +42,24 @@ def init_rank(num_ranks, uid, rank):
     return torch._C._nccl_init_rank(num_ranks, uid, rank)
 
 
+def _check_sequence_type(inputs):
+    if not isinstance(inputs, list) and not isinstance(inputs, tuple):
+        raise TypeError("inputs should be a list/tuple")
```

Comment:
Can it also be any python iterable/list-like object? I just tried the NCCL test with `inputs = iter(inputs)` (which makes it a `list_iterable` type) and `inputs = set(inputs)`, and the current `test_all_gather` seems to pass. With this PR, would we stop supporting those types?